model/src/pyrenew/metaclass.py

```diff
@@ -887,6 +889,19 @@ def sample(self, **kwargs) -> tuple:
             for t, uv in zip(self.transforms, untransformed_values)
         )
 
+        if record:
+            if hasattr(untransformed_values, "_fields"):
+                for i, tv in enumerate(transformed_values):
+                    suffix = untransformed_values._fields[i]
+                    numpyro.deterministic(f"{self.name}_{suffix}", tv.value)
+            elif len(untransformed_values) == 1:
+                numpyro.deterministic(self.name, transformed_values[0].value)
+            else:
+                for i, tv in enumerate(transformed_values):
+                    numpyro.deterministic(f"{self.name}_{i}", tv.value)
+
+        return transformed_values
+
```

Comment:
You can make this more compact by computing the `suffix` as either `untransformed_values._fields[i]` or `i` depending on ` hasattr(untransformed_values, "_fields")`, and then use a single `for i, tv in enumerate(transformed_values):` block.